src/GraphQL/GraphQLExtensions.cs

```diff
@@ -76,6 +76,16 @@ public static bool IsInputObjectType(this IGraphType type)
             return namedType is IInputObjectGraphType;
         }
 
+        /// <summary>
+        /// Determines if this graph type is an introspection type
+        /// </summary>
+        public static bool IsIntrospectionType(this IGraphType graphType) => graphType?.Name?.StartsWith("__") ?? false;
+
+        /// <summary>
+        /// Determines if this field is an introspection type
+        /// </summary>
+        public static bool IsIntrospectionField(this FieldType fieldType) => fieldType?.Name?.StartsWith("__") ?? false;
```

Comment:
Well, is `types` field or any other child field in `__schema` introspection field or not?

Reply:
It is possible that to solve the problem, the time has come to do what I have long wanted - to make a reference from `FieldType` to `IGraphType`, `Parent` property. I needed this property to solve other problems. If it were, then the question of whether the field belongs to introspection would become trivial. In addition, the conditions inside `TryGetArgument`/`HasArgument` will be simplified. I believe that now these conditions are not correct because they do not take into account nested introspection fields. The converter should not be applied to the entire introspection tree.

I believe that the task of adding `Parent` property to `FieldType` is more important to solve in separate PR first. Then we can return to this PR and do things simpler. What do you think?

Reply:
I'm not sure exactly how that is implemented in the code, but I did verify that both user field names and user argument names must not start with `__`.  So the check I've implemented, pursuant to your previous suggestion, should be a valid one.

Because I think this is a valid way to handle this issue, I'd pull it in, which will fix two existing bugs in the code base.  (a) introspection arguments not working with pascal case, and (b) HasArgument not converting names via the selected name converter.  It will also implement the new interface design we have come up with.

Once `FieldType.Parent` is added, we can update this code to use a more elegant solution, as it should not change the work already accomplished.

As it relates to your idea, this method is checking for an introspection query field "on the `Query` type", rather than a field as a child of an introspection type.  Would `FieldType.Parent` point to the Query graph, or a constant, or null?  Is this `Parent` field an `IGraphType` or a `Type`?  I'm already checking `context.ParentType` for `null` to determine if this might be an introspection query field (see `GetArgument`), so depending on the decisions to the above answers, it may or may not help this code at all.

In general, I like the idea of adding `Parent` to `FieldType`.  I just feel that there's some unanswered questions about how this will play out, and whatever the solution, it should not affect the changes to `IFieldNameConverter` that constitutes the majority of this PR - but rather a simple optimization (if any) of the `IsIntrospectionField` or `IsIntrospectionType` methods.  We could make those two new methods `internal` if we think the definitions may change soon -- there's no real reason they need to be public at this time.

Reply:
> Well, is types field or any other child field in __schema introspection field or not?

I'm sorry, I don't understand your question.  I hope I clarified what the two new methods do in my previous comment.  Perhaps I should update the xml comments to more clearly define what the methods do, as I admit it's not that clear.

Reply:
```graphql
__schema (1)
{
   types   (2)
   {
      name  (3)
      kind   (4)
   }
}
```

IsIntrospectionField(1) ?
IsIntrospectionField(2) ?
IsIntrospectionField(3) ?
IsIntrospectionField(4) ?

Following the logic of the `IsIntrospectionField` method, will the names of the fields be converted in each of the four cases? Should a conversion take place?

Reply:
> I'm not sure exactly how that is implemented in the code, but I did verify that both user field names and user argument names must not start with __.

This is a **necessary** condition, but **not sufficient**.

Reply:
At present, and by design, only 1 returns true, as it is the only introspection field [on a non-introspection type].

2, 3, and 4 are covered by IsIntrospectionType, checking the parent type of the field.

GetArgument only executes either IsIntrospectionType or IsIntrospectionField, not both, depending on if parent is null.

Reply:
Conversions upon schema initialization is handled as before, where they are all converted by the camel case converter, no matter what converter was selected for the schema.  This is necessary as some of the fields are defined in pascal case.

Reply:
Damn, I forgot about it again. 🤦‍♂

Reply:
I again thought about the unnecessary side effects of using a camel case converter on the introspection fields. And again I forgot that these effects are actually useful. I will continue the review tomorrow.

Reply:
Ok no problem.

Reply:
So, I review your changes again and I think that both `IsIntrospectionType` and `IsIntrospectionField` methods should be `internal`. In fact, as discussed above, they work only for some introspection fields and their names can be misleading. It will be even better to move these methods inside `ResolveFieldContextExtensions` and make them `private`.

Reply:
> So, I review your changes again and I think that both `IsIntrospectionType` and `IsIntrospectionField` methods should be `internal`. In fact, as discussed above, they work only for some introspection fields and their names can be misleading. It will be even better to move these methods inside `ResolveFieldContextExtensions` and make them `private`.

Done; with updated xml comment